### PR TITLE
fix: avoid problem during tree data extraction

### PR DESCRIPTION
**Pull request by veinfors**
Additional checks to avoid problem during data extraction for tree data nodes that are missing qAttrDims or qAttrExps.

### CODE REVIEW

**Comment by T-Wizard** [comment]
when is this needed?

**Comment by veinfors** [comment]
> when is this needed?

When fetching data using `GetHyperCubeTreeData` and there is a data set which doesn't contain all possible dimension values in all "parent" nodes it's possible to send in a parameter called `AllValues` to engine. This tells engine to "pad"/"expand" with all possible values from the other dimension. But when doing this, engine is (of course) not adding values or attribute dimensions or attribute expressions on those nodes, since they don't exist. This PR makes it possible to use that kind of data set with picasso without it crashing.

**Review by T-Wizard — APPROVED** [review]

File: plugins/q/src/data/__tests__/extractor-t.spec.js [hunk 1]
```diff
@@ -914,6 +914,56 @@ describe('q-data-extractor-t', () => {
...
+      expect(m).to.eql([
+        { value: NaN, label: '', source: { key: 'cube', field: 'qDimensionInfo/1/qAttrExprInfo/0' } },
+        { value: 2, label: 'two', source: { key: 'cube', field: 'qDimensionInfo/1/qAttrExprInfo/0' } },
+        { value: 5, label: 'five', source: { key: 'cube', field: 'qDimensionInfo/1/qAttrExprInfo/0' } },
+        { value: 7, label: 'seven', source: { key: 'cube', field: 'qDimensionInfo/1/qAttrExprInfo/0' } },
+        { value: 9, label: 'nine', source: { key: 'cube', field: 'qDimensionInfo/1/qAttrExprInfo/0' } },
+      ]);
+
+      // Restore
+      treePages[0].qNodes[0].qNodes[0].qAttrExps = tempQAttrExps;
```
Comment (cbt1) [inline plugins/q/src/data/__tests__/extractor-t.spec.js:965]: This wont be called if the `expect` ☝️ fails

**Review by cbt1 — CHANGES_REQUESTED** [review]
Fix restore

File: plugins/q/src/data/__tests__/extractor-t.spec.js (hunk 1 above)
Comment (veinfors) [reply plugins/q/src/data/__tests__/extractor-t.spec.js]: Good point! Fixed!

**Review by cbt1 — APPROVED** [review]

File: plugins/q/src/data/__tests__/extractor-t.spec.js (hunk 1 above)
Comment (T-Wizard) [reply plugins/q/src/data/__tests__/extractor-t.spec.js]: grouping the test with a describe and moving this to an after/afterEach function will also run it if there is a error during the extract call.

**Pull request merged**